git/provider_gitlab.go

```diff
@@ -84,31 +84,6 @@ func (p *GitLabProvider) CreateRepository(ctx context.Context, r *Repository) (b
 	return true, nil
 }
 
-// GetRepositoryOwner returns the actual path owner. This is need for Gitlab where the name of a group might differ
-// from its path
-func (p *GitLabProvider) GetRepositoryOwner(ctx context.Context, token string, owner string) (string, error) {
-	gl, err := gitlab.NewClient(token)
-	if err != nil {
-		return "", fmt.Errorf("client error: %w", err)
-	}
-
-	groupName := strings.Split(owner, "/")[0]
-	lgo := &gitlab.ListGroupsOptions{
-		Search:         gitlab.String(groupName),
-		MinAccessLevel: gitlab.AccessLevel(gitlab.GuestPermissions),
-	}
-	groups, _, err := gl.Groups.ListGroups(lgo, gitlab.WithContext(ctx))
-	if err != nil {
-		return "", fmt.Errorf("failed to list groups, error: %w", err)
-	}
-
-	if len(groups) == 0 {
-		return "", fmt.Errorf("failed to find group named '%s'", groupName)
-	}
-
-	return groups[0].Path, nil
-}
```

Comment:
As this interface is public and we want to be nice to our package consumers (even if that's just us), I think it would be good practice to keep this for the time being, and add a `Deprecated` note to the doc block like this:

```
// Deprecated, has become obsolete due to changes in AddTeam.
// 
// GetRepositoryOwner returns the actual path owner. This is need for Gitlab where the name of a group might differ
// from its path
```

(or something that's more descriptive).